src/v/schema/registry.cc

```diff
@@ -43,6 +43,12 @@ class schema_registry_impl : public registry {
         co_return reader;
     }
 
+    ss::future<ppsr::schema_getter*> synced_getter() const override {
+        auto [reader, writer] = co_await service();
+        co_await writer->read_sync();
+        co_return reader;
+    }
```

Comment:
>do this conditionally because the sync seems somewhat expensive.

not to split hairs, but is there an actual case where we want the faster, unsynced behavior?

Reply:
Do you mean an actual case for unsynced `get_valid_schema()`? I can't think of one, and we're the only one using this API right now.

Or if you're referring to unsynced `getter()`, the case for a non-synced `getter()` is that it allows us to implement the cheap condition in `get_valid_schema()`

Reply:
ah, i should have expanded `get_valid_schema` in the diff. first I'm looking at this wrapper. got it, thanks.